single-host-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/singlehostclient/SingleHostHttpClientBuilder.java

```diff
@@ -43,8 +46,8 @@ private SingleHostHttpClientBuilder(String hostname, HttpClient.Builder builder)
      * It also provides the given hostname in SNI extension.
      * The returned java.net.http.HttpClient is wrapped to force the HTTP header <code>Host</code> with the given hostname.
      */
-    public static HttpClient newHttpClient(String hostname) {
-        return builder(hostname, HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(2L)))
+    public static HttpClient newHttpClient(String hostname, InetAddress hostAddress) {
+        return builder(hostname, hostAddress, HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(2L)))
```

Comment:
_A **side** and **offtopic** note_, but I am still thinking we have to make this `2L` seconds constant configurable.

Reply:
It's a reasonable default when the `HttpClient` is not explicitly provided. Default from the client is infinite which clearly breaks the resilient client.